torchmdnet/priors/coulomb.py

```diff
@@ -0,0 +1,50 @@
+import torch
+from torchmdnet.priors.base import BasePrior
+from torchmdnet.models.utils import Distance
+from torch_scatter import scatter
+from typing import Optional, Dict
+
+class Coulomb(BasePrior):
+    """This class implements a Coulomb potential, scaled by erf(alpha*r) to reduce its
+    effect at short distances.
+
+    To use this prior, the Dataset must include a field called `partial_charges` with each sample,
+    containing the partial charge for each atom.  It also must provide the following attributes.
+
+    distance_scale: multiply by this factor to convert coordinates stored in the dataset to meters
+    energy_scale: multiply by this factor to convert energies stored in the dataset to Joules (*not* J/mol)
+    """
+    def __init__(self, alpha, max_num_neighbors, distance_scale=None, energy_scale=None, dataset=None):
+        super(Coulomb, self).__init__()
+        if distance_scale is None:
+            distance_scale = dataset.distance_scale
+        if energy_scale is None:
+            energy_scale = dataset.energy_scale
+        self.distance = Distance(0, torch.inf, max_num_neighbors=100)
```

Comment:
Why are you hard coding `max_num_neighbors` instead of using the parameter?

Reply:
Thanks for catching that.  That was just done for development.  I meant to create a parameter for it but forgot.

Reply:
Fixed.